components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java

```diff
@@ -123,7 +124,7 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
                     log.debug("A registered IdP was not found the given realm");
                 }
             }
-        } catch (IdentityApplicationManagementException e) {
+        } catch (IdentityProviderManagementException e) {
             log.error("Exception while getting IdP by realm", e);
```

Comment:
Ideally we must be throwing this exception and its the client who can decide what to do with it. Lets change this to throw the exception and in authentication framework we can catch the exception, log it and continue.